OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -63,6 +64,24 @@ public class CloakInfo : PausableConditionalTraitInfo
 		[Desc("The condition to grant to self while cloaked.")]
 		public readonly string CloakedCondition = null;
 
+		[Desc("Which image to use for the effect played when cloaking or uncloaking.")]
+		public readonly string EffectImage = null;
+
+		[Desc("Which effect sequence to play when cloaking.")]
+		[SequenceReference(nameof(EffectImage), allowNullImage: true)]
+		public readonly string CloakEffectSequence = null;
+
+		[Desc("Which effect sequence to play when uncloaking.")]
+		[SequenceReference(nameof(EffectImage), allowNullImage: true)]
+		public readonly string UncloakEffectSequence = null;
+
+		[PaletteReference(nameof(EffectPaletteIsPlayerPalette))]
+		public readonly string EffectPalette = "effect";
+		public readonly bool EffectPaletteIsPlayerPalette = false;
```

Comment:
This isn't ever used by the code, so if your mod needs this you need to use it too. :)

Reply:
oh damn, sorry for that

Reply:
![sparkle](https://user-images.githubusercontent.com/13763394/149624944-aa474082-4752-490d-8da0-b952d4c11e87.gif)

Hope you like those sparkles

Reply:
fixed and I also need to fix TA/RV engine on this :(